omod-1.9/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_9/FrontendJsonConfigController1_9.java

```diff
@@ -0,0 +1,128 @@
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.apache.commons.io.IOUtils;
+import org.codehaus.jackson.JsonProcessingException;
+import org.codehaus.jackson.map.ObjectMapper;
+
+@Controller
+@RequestMapping(value = "/rest/" + RestConstants.VERSION_1 + "/frontend/config.json")
+public class FrontendJsonConfigController1_9 extends BaseRestController {
+
+    private static final String DEFAULT_FRONTEND_DIRECTORY = "frontend";
+    private static final String GP_LOCAL_DIRECTORY = "spa.local.directory";
+    private static final String JSON_CONFIG_FILE_NAME = "config.json";
+    private static final Logger log = LoggerFactory.getLogger(FrontendJsonConfigController1_9.class);
+
+    @RequestMapping(method = RequestMethod.GET)
+	public void getFrontendConfigFile(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        File jsonConfigFile = getJsonConfigFile();
+        if (!jsonConfigFile.exists()) {
+            log.debug("Configuration file does not exist");
+            response.sendError(HttpServletResponse.SC_NOT_FOUND, "Configuration file does not exist");
+        }
+        try {
+            InputStream inputStream = new FileInputStream(jsonConfigFile);
+            OutputStream outStream = response.getOutputStream();
+            OpenmrsUtil.copyFile(inputStream, outStream);
+    
+            response.setContentType("application/json");
+            response.setHeader("Content-Disposition", "attachment; filename=" + jsonConfigFile.getName());
+            response.setStatus(HttpServletResponse.SC_OK);
+        } catch (IOException e) {
+            log.error("Error reading Configuration file: " + jsonConfigFile.getAbsolutePath(), e);
+            response.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR, "Error reading Configuration file: " + jsonConfigFile.getPath());
+        }
+	}
+
+    @RequestMapping(method = RequestMethod.POST)
+    @ResponseStatus(HttpStatus.OK)
+    public void saveFrontendConfigFile(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        User user = Context.getAuthenticatedUser();
+        if (user == null || !user.isSuperUser()) {
+            log.error("Authorization error while creating a config.json file");
+            response.sendError(HttpServletResponse.SC_FORBIDDEN, "Authorization error. Admin privileges required to save the config.json file");
+            return;
+        }
+        saveJsonConfigFile(request, response);
+    }
+
+    private void saveJsonConfigFile(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        File jsonConfigFile = getJsonConfigFile();
+        InputStream inputStream = request.getInputStream();
+        String requestBody = IOUtils.toString( inputStream , "UTF-8");
```

Comment:
Is the above line also for validation?

Reply:
no, this is just for reading the request body.  Line 86 does the validation